src/data/set/intervals/basic.lean

```diff
@@ -365,6 +365,33 @@ begin
     apply_rules [subset_diff_singleton] }
 end
 
+lemma mem_Ioo_or_eq_endpoints_of_mem_Icc {x : α} (hmem : x ∈ Icc a b) :
+  x = a ∨ x = b ∨ x ∈ Ioo a b :=
+begin
+  rw [mem_Icc, le_iff_lt_or_eq, le_iff_lt_or_eq] at hmem,
+  rcases hmem with ⟨hxa | hxa, hxb | hxb⟩,
+  exact or.inr (or.inr ⟨hxa, hxb⟩),
```

Comment:
When you have several goals, each proof of a goal should be enclosed in braces. Same thing below.